modules/private/rebuild_indexes.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2019 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package private
+
+import (
+	"fmt"
+
+	"code.gitea.io/gitea/modules/setting"
+)
+
+// RebuildRepoIndex rebuild a repository index
+func RebuildRepoIndex(repoID int64) (bool, error) {
+	// Ask for running deliver hook and test pull request tasks.
+	reqURL := setting.LocalURL + fmt.Sprintf("api/internal/maint/rebuild-repo-index/%d", repoID)
```

Comment:
So that when gitea is not running, you can not rebuild the repo index?

Reply:
> So that when gitea is not running, you can not rebuild the repo index?

@lunny That's correct. I had to choose between using the indexing queue for the task (requires Gitea running) or deleting the files/records manually (requires Gitea to be not running).

After all this work I think I might have went for the second option. However, the first one is more future proof because it will work well if we add Gitea the capability to change indexing options on the fly on a per repository basis. We can then rebuild the indexes when the user change those options.